armi/reactor/converters/uniformMesh.py

```diff
@@ -85,6 +86,317 @@ def converterFactory(globalFluxOptions):
+        """
+        assemblyTypeFlags = set(
+            [a.p.flags for a in self._sourceReactor.core.getAssemblies(Flags.FUEL)]
+        )
+        fuelBottoms = [
+            min(
+                round(
+                    a.getFirstBlock(Flags.FUEL).p.zbottom, self.FLOAT_ROUNDING_DECIMALS
+                )
+                for a in self._sourceReactor.core.getAssemblies(assemFlags, exact=True)
+            )
+            for assemFlags in assemblyTypeFlags
+        ]
+        fuelTops = [
+            max(
+                round(a.getBlocks(Flags.FUEL)[-1].p.ztop, self.FLOAT_ROUNDING_DECIMALS)
+                for a in self._sourceReactor.core.getAssemblies(assemFlags, exact=True)
+            )
+            for assemFlags in assemblyTypeFlags
+        ]
+        filteredBottoms = self._filterMesh(
+            fuelBottoms,
+            self.minimumMeshSize,
+            [min(fuelBottoms)],
+            preference="bottom",
+            warn=True,
+        )
+        filteredTops = self._filterMesh(
+            fuelTops, self.minimumMeshSize, [max(fuelTops)], preference="top", warn=True
+        )
+
+        return filteredBottoms, filteredTops
+
+    def _getFilteredControlTopAndBottom(self, fuelBottoms, fuelTops):
+        """
+        Get the bottom and top boundaries of control assemblies filtered to the ``minimumMeshSize``.
+
+        Returns
+        -------
+        filteredBottomCtrl : the bottom of control assemblies, filtered to a minimum separation of
+            ``minimumMeshSize`` with preference for the lowest bounds
+        filteredTopCtrl : the top of control assemblies, filtered to a minimum separation of
+            ``minimumMeshSize`` with preference for the top bounds
+        """
+        bottomMatBoundaries = set(fuelBottoms)
+        topMatBoundaries = set(fuelTops)
+
+        # find all control assembly boundaries
+        for a in self._sourceReactor.core.getAssemblies(Flags.CONTROL):
+            firstBlock = a.getFirstBlock(Flags.CONTROL)
+            lastBlock = a.getBlocks(Flags.CONTROL)[-1]
+            bottomMatBoundaries.add(
+                round(firstBlock.p.zbottom, self.FLOAT_ROUNDING_DECIMALS)
+            )
+            topMatBoundaries.add(round(lastBlock.p.ztop, self.FLOAT_ROUNDING_DECIMALS))
+
+        bottomBoundList = sorted(list(bottomMatBoundaries))
+        topBoundList = sorted(list(topMatBoundaries))
+        # filter control boundaries to minimum mesh size
+        filteredBottomCtrl = self._filterMesh(
+            bottomBoundList,
+            self.minimumMeshSize,
+            fuelBottoms,
+            preference="bottom",
+            warn=True,
+        )
+        filteredTopCtrl = self._filterMesh(
+            topBoundList, self.minimumMeshSize, fuelTops, preference="top", warn=True
+        )
+        return filteredBottomCtrl, filteredTopCtrl
+
+    @staticmethod
+    def _determineIndexToRemove(ileft, iright, meshPoints, anchorPoints):
```

Comment:
I think I would prefer this logic back in the original function...
instead of 
```
                    removeIndex = self._determineIndexToRemove(
                        i, i + 1, meshList[i : i + 2], anchorPoints
                    )
```


It would be 
```
                    if meshList[i] in anchorPoints and meshList[i+1] in anchorPoints:
                        raise ValueError
                    if meshList[i] in anchorPoints:
                        removeIndex = i
                    if meshList[i+1] in anchorPoints:
                        removeIndex = i +1
```
would would seem more clear to me. The logic here does not seem complicated enough that hit needs a separate static method.